shell/platform/android/android_platform_surface_gl.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/android/android_platform_surface_gl.h"
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES/gl.h>
+#include <GLES/glext.h>
+#include <jni.h>
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/resource_context.h"
+#include "flutter/shell/platform/android/platform_view_android_jni.h"
+#include "third_party/skia/include/core/SkSurface.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrTexture.h"
+#include "third_party/skia/include/gpu/GrTypes.h"
+
+namespace shell {
+
+AndroidPlatformSurfaceGL::~AndroidPlatformSurfaceGL() {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch]() {
+    glDeleteTextures(1, &texture_id_);
+    latch.Signal();
+  });
+  latch.Wait();
+}
+
+AndroidPlatformSurfaceGL::AndroidPlatformSurfaceGL() {
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch]() {
+    GrGLuint texID;
+    glGenTextures(1, &texID);
+
+    glBindTexture(GL_TEXTURE_EXTERNAL_OES, texID);
+    texture_id_ = texID;
+    latch.Signal();
+  });
+  latch.Wait();
+}
+
+void AndroidPlatformSurfaceGL::MarkNewFrameAvailable() {
+  ASSERT_IS_PLATFORM_THREAD;
+  blink::Threads::IO()->PostTask([this]() { new_frame_ready_ = true; });
+}
+
+sk_sp<SkImage> AndroidPlatformSurfaceGL::MakeSkImage(int width,
+                                                     int height,
+                                                     GrContext* grContext) {
+  ASSERT_IS_GPU_THREAD;
+  fxl::AutoResetWaitableEvent latch;
+  blink::Threads::IO()->PostTask([this, &latch]() {
```

Comment:
I want to echo the same concerns I raised in the iOS variant of this call here. You are already on the GPU thread which has an EGL context in the same sharegroup as the IO thread EGL context. Use the context you have already instead of hopping threads.